contracts/MintBurnERC20.sol

```diff
@@ -0,0 +1,77 @@
+//SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "./access/AdminRole.sol";
+
+/**
+ * @dev Collection of functions related to the address type
+ */
+library AddressLib {
+    /**
+     * @dev Returns true if `account` is a contract.
+     */
+    function isContract(address account) internal view returns (bool) {
+        // This method relies on extcodesize, which returns 0 for contracts in
+        // construction, since the code is only stored at the end of the
+        // constructor execution.
+
+        uint256 size;
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            size := extcodesize(account)
+        }
+        return size > 0;
+    }
+}
+
+contract MintBurnERC20 is ERC20, AdminRole {
+    using AddressLib for address;
+    uint8 private _decimals = 18;
+    address private _safe;
+
+    constructor(string memory tokenName, string memory tokenSymbol, uint8 numDecimals) ERC20(tokenName, tokenSymbol) {
+        _decimals = numDecimals;
+    }
+
+    function transferFrom(address sender, address recipient, uint256 amount) public virtual override returns (bool) {
+        if (sender == _safe) {
```

Comment:
The safe should never call transferFrom from himself. transferFrom is called only on deposits. We can either omit this check to save gas or we can revert

Reply:
removed